### PR TITLE
Show touches option.

### PR DESCRIPTION
**Pull request by monxalo**
This option turns on 'Show touches' on the Developer Options when recording. Also needs
a new permission to change system settings.

### CODE REVIEW

File: telecine/src/main/java/com/jakewharton/telecine/TelecineService.java [hunk 1]
```diff
@@ -35,6 +37,10 @@ public static Intent newIntent(Context context, int resultCode, Intent data) {
 
   private final RecordingSession.Listener listener = new RecordingSession.Listener() {
     @Override public void onStart() {
+      if(showTouchesProvider.get()) {
+        Settings.System.putInt(TelecineService.this.getContentResolver(), "show_touches", 1);
```
Comment (monxalo) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineService.java]: I really don't like using `Telecine.this` reference but this location seemed apropriate for this case. If you have another sugestion, i would gladly change it.

**Comment by jaredsburrows** [comment]
@monxalo @JakeWharton  What is the status on this?

**Comment by monxalo** [comment]
@jaredsburrows This is done and working, at least on my device.

**Comment by jaredsburrows** [comment]
@monxalo  I see you just committed again. Are you just waiting for @JakeWharton to accept? He may not reply if this was not an "issue" that was listed or created.

**Comment by monxalo** [comment]
@jaredsburrows I was just waiting for some feedback. I'll create an issue for this then.

**Comment by jaredsburrows** [comment]
@monxalo I agree with your PR, I wish I could merge it, I am not a collaborator :(.

**Comment by NightlyNexus** [comment]
Settings.System.putInt(contentResolver, "show_touches", 0);
does not seem to be documented.

File: telecine/src/main/java/com/jakewharton/telecine/TelecineService.java [hunk 2]
```diff
@@ -27,14 +29,20 @@ public static Intent newIntent(Context context, int resultCode, Intent data) {
...
 
   @Inject Analytics analytics;
+  @Inject ContentResolver contentResolver;
 
   private boolean running;
   private RecordingSession recordingSession;
 
   private final RecordingSession.Listener listener = new RecordingSession.Listener() {
     @Override public void onStart() {
+      if(showTouchesProvider.get()) {
```
Comment (JakeWharton) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineService.java]: nit: space after `if`

File: telecine/src/main/java/com/jakewharton/telecine/TelecineService.java [hunk 3]
```diff
@@ -56,6 +64,10 @@ public static Intent newIntent(Context context, int resultCode, Intent data) {
     }
 
     @Override public void onStop() {
+      if(showTouchesProvider.get()) {
```
Comment (JakeWharton) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineService.java]: nit: space after `if`

File: telecine/src/main/java/com/jakewharton/telecine/TelecineService.java [hunk 4]
```diff
@@ -56,6 +64,10 @@ public static Intent newIntent(Context context, int resultCode, Intent data) {
...
+        Settings.System.putInt(contentResolver, "show_touches", 0);
```
Comment (JakeWharton) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineService.java]: This string should be pulled into a static final constant.

File: telecine/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -110,4 +110,24 @@
         />
   </LinearLayout>
 
+  <LinearLayout
+    android:layout_width="match_parent"
+    android:layout_height="48dp"
+    android:orientation="horizontal"
+    android:gravity="center_vertical"
+    >
+    <TextView
```
Comment (JakeWharton) [inline telecine/src/main/res/layout/activity_main.xml]: Tags should be indented with two spaces.

File: telecine/src/main/java/com/jakewharton/telecine/TelecineService.java (hunk 2 above)
Comment (jaredsburrows) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineService.java]: @monxalo  Auto format with intellij

File: telecine/src/main/java/com/jakewharton/telecine/TelecineService.java (hunk 4 above)
Comment (jaredsburrows) [inline telecine/src/main/java/com/jakewharton/telecine/TelecineService.java]: +1

File: telecine/src/main/res/layout/activity_main.xml (hunk 5 above)
Comment (jaredsburrows) [inline telecine/src/main/res/layout/activity_main.xml]: Auto format with intellij

**Comment by monxalo** [comment]
@NightlyNexus yes, it's not documented. You can check these keys on the source code of the Apps Settings [here](https://github.com/android/platform_packages_apps_settings/blob/master/src/com/android/settings/DevelopmentSettings.java).

**Comment by NightlyNexus** [comment]
@monxalo Right, I was just noting that it is not a guarantee.  Ah, I did not realize that it will only silently fail if the string does not match, though.  Anyway, I like the addition.

**Comment by jaredsburrows** [comment]
This should be good to go now :)

**Comment by jaredsburrows** [comment]
@monxalo @JakeWharton Anything else left?

**Pull request merged**